registry.go

```diff
@@ -1,20 +1,36 @@
 package distribution
 
-import (
-	"io"
-	"time"
+import "golang.org/x/net/context"
 
-	"github.com/docker/distribution/digest"
-	"github.com/docker/distribution/manifest"
-	"golang.org/x/net/context"
-)
+// Scope defines the set of items that match a namespace.
+type Scope interface {
+	// Contains returns true if the name belongs to the namespace.
+	Contains(name string) (bool, error)
```

Comment:
We can probably remove this. It really depends on how we define scope. If it is just a syntax-based set (ie, glob or equiv), then we don't need the error here.

What if the implementation has to ask a registry?